src/vs/workbench/contrib/interactiveSession/browser/actions/interactiveSessionCodeblockActions.ts

```diff
@@ -97,28 +98,42 @@ export function registerInteractiveSessionCodeBlockActions() {
 			const editorService = accessor.get(IEditorService);
 			const bulkEditService = accessor.get(IBulkEditService);
 			const interactiveSessionService = accessor.get(IInteractiveSessionService);
-			const activeEditorControl = editorService.activeTextEditorControl;
-			if (isCodeEditor(activeEditorControl)) {
-				const activeModel = activeEditorControl.getModel();
-				if (!activeModel) {
-					return;
-				}
+			const textFileService = accessor.get(ITextFileService);
+
+			let activeEditorControl = editorService.activeTextEditorControl;
+			if (isDiffEditor(activeEditorControl)) {
+				activeEditorControl = activeEditorControl.getOriginalEditor().hasTextFocus() ? activeEditorControl.getOriginalEditor() : activeEditorControl.getModifiedEditor();
+			}
+
+			if (!isCodeEditor(activeEditorControl)) {
+				return;
+			}
 
-				const activeSelection = activeEditorControl.getSelection() ?? new Range(activeModel.getLineCount(), 1, activeModel.getLineCount(), 1);
-				await bulkEditService.apply([new ResourceTextEdit(activeModel.uri, {
-					range: activeSelection,
-					text: context.code,
-				})]);
-
-				interactiveSessionService.notifyUserAction(<IInteractiveSessionUserActionEvent>{
-					providerId: context.element.providerId,
-					action: {
-						kind: 'insert',
-						responseId: context.element.providerResponseId,
-						codeBlockIndex: context.codeBlockIndex,
-					}
-				});
+			const activeModel = activeEditorControl.getModel();
+			if (!activeModel) {
+				return;
+			}
+
+			const activeTextModel = textFileService.files.get(activeModel.uri);
```

Comment:
I don't quite understand this? Is the point of this just the `isReadonly` check? And do you want to use activeTextModel instead of activeModel after this?

Reply:
Yes I think only `ITextFileEditorModel` has a `readonly` property. Not sure if there's a better way to check this without using `ITextFileService`